addons/sale/models/sale_order_line.py

```diff
@@ -489,7 +489,7 @@ def _compute_product_uom(self):
             if not line.product_uom or (line.product_id.uom_id.id != line.product_uom.id):
                 line.product_uom = line.product_id.uom_id
 
-    @api.depends('product_id', 'company_id')
+    @api.depends('product_id', 'company_id', 'order_id.payment_term_id')
```

Comment:
I don't understand why you need this extra dependency

Reply:
So that the tax amount si re-computed correctly when changing the customer or adding the product before changing the payment term (realized it didn't work after a test)

Reply:
Seen like this it is indeed not ideal. But changing the payment term and the tax not being recomputed is an incorrect display and afterwards an incorrect computation. Do you have another suggestion ?

Reply:
We did introduce an 'Update Prices' for the pricelist, and an 'Update Taxes' for the fiscal position, maybe propose a similar mechanism for the payment terms ?

Reply:
I'll have a look !

Reply:
Still on the having-a-look stage, I want to say however that this behavior (recomputing when changing the payment term) is what occurs in invoices and is intended. 
Don't we want to align those behaviors ?

Reply:
Not so sure we want, at least not through the compute.
Sales orders (even more now that some can be subscription orders) tend to be more long-lived than invoices, and we don't want a partner modification to recompute everything on the SO (we had the issue multiple times in the past, including in internal where they do shady stuff).  As the ORM triggers the recomputation of fields even if their none of their dependencies changed (but one or more was recomputed), the `tax_id` would be recomputed on any partner changed, which is not what we want.

Through an onchange, it could be considered, but even that approach could be risky.
It's globally safer for the user to be aware of the implications of their change(s), and only do it if necessary.

Reply:
Following this I just removed the depends. If we realise later on this causes a problem for the user it will be an easy fix, if it doesn't we will know it wasn't needed.